cmd/frontend/internal/frontend/service.go

```diff
@@ -288,95 +293,79 @@ func (fes *FrontEndService) Monitor(ctx context.Context, errCh chan<- error) {
+			if err != nil {
+				logger.Error(err, "BFD output", "out", strings.Split(bfdOut, "\n"))
+				denounce = true
+				continue
+			}
+
+			// determine status of gateways and connectivity
+			status := fes.parseStatusOutput(protocolOut, bfdOut)
+
+			// Gateway availibility notifications
+			if status.NoConnectivity() {
+				// although configured at least one IP family has no connectivity
+				if hasConnectivity {
+					denounce = true
+					// should this be moved to "if denounce" above?
+					health.SetServingStatus(ctx, health.EgressSvc, false)
+					fes.denounceVIP(ctx, errCh)
 				}
 			} else {
-				bfdOut, err := fes.routingService.ShowBfdSessions(ctx, lp, `NBR-BFD`)
-				if err != nil {
-					logger.Info("BFD output", "err", err, "out", strings.Split(bfdOut, "\n"))
-					//Note: if birdc is not yet running, no need to bail out
-					//linkCh <- "Failed to fetch bfd status"
-					break
-				}
-				// determine status of gateways and connectivity
-				status := fes.parseStatusOutput(protocolOut, bfdOut)
-
-				// Gateway availibility notifications
-				// XXX: in case denounceFrontend/announceFrontend would block the thread for too long
-				// (no NSP is listening etc.), and it is a problem, move them to dedicated go thread
-				// TODO: maybe move logic to separate function
-				if status.NoConnectivity() {
-					// although configured at least one IP family has no connectivity
-					// Note: deanounce FE even if just started (init); container might have crashed
-					if !noConnectivity || init {
-						noConnectivity = true
-						health.SetServingStatus(ctx, health.EgressSvc, false)
-						if refreshCancel != nil {
-							refreshCancel()
-						}
-						if err := denounceFrontend(fes.targetRegistryClient); err != nil {
-							logger.Error(err, "Denounce frontend connectivity")
-						}
-						fes.denounceVIP(ctx, errCh)
-					}
-				} else {
-					if noConnectivity {
-						noConnectivity = false
-						health.SetServingStatus(ctx, health.EgressSvc, true)
-						logger.Info("Announce frontend")
-						if err := announceFrontend(fes.targetRegistryClient); err != nil {
-							logger.Error(err, "Announce frontend connectivity")
-						}
-						// refresh NSP entry
-						var refreshCtx context.Context
-						refreshCtx, refreshCancel = context.WithCancel(ctx)
-						go func() {
-							_ = retry.Do(func() error {
-								return announceFrontend(fes.targetRegistryClient)
-							}, retry.WithContext(refreshCtx),
-								retry.WithDelay(fes.nspEntryTimeout),
-								retry.WithErrorIngnored())
-						}()
-						fes.announceVIP(ctx, errCh)
+				if !hasConnectivity {
+					hasConnectivity = true
+					health.SetServingStatus(ctx, health.EgressSvc, true)
+					logger.Info("Announce frontend")
+					if err := announceFrontend(ctx, fes.targetRegistryClient); err != nil {
+						logger.Error(err, "Announce frontend connectivity")
 					}
+					// refresh NSP entry (even if announceFrontend() above fails?)
```

Comment:
Yes, even if the send towards NSP fails, the FE still has external connectivity. The FE should keep trying to inform NSP about it.

Reply:
That question is still in place. Would be better to remove, in order to avoid confusion.
Without the retry block, the FE would not be able to announce its connectivity without a prior state change (-> down, up).